tests/test_clients_AwsClient.py

```diff
@@ -0,0 +1,139 @@
+import os
+import pytest
+from unittest.mock import patch
+import boto3
+from botocore.config import Config
+from lib.clients.AwsClient import AwsClient
+from lib.clients.BaseClient import BaseClient
+
+#Test data
+valid_container = 'backup-container'
+invalid_container = 'invalid-container'
+configuration = {
+    'credhub_url' : None,
+    'type' : 'online',
+    'backup_guid' : 'backup-guid',
+    'instance_id' : 'vm-id',
+    'secret' : 'xyz',
+    'job_name' : 'service-job-name',
+    'container' : valid_container,
+    'access_key_id' : 'key-id',
+    'secret_access_key' : 'secret-key',
+    'region_name' : 'xyz'
+}
+
+directory_persistent = '/var/vcap/store'
+directory_work_list = '/tmp'
+log_dir = 'tests'
+poll_delay_time = 10
+poll_maximum_time = 60
+operation_name = 'backup'
+availability_zone = 'abc'
+
+# Defintions of dummy clients and resources.
+# More details to be added as new tests will be added 
+class Ec2ConfigDummy:
+    def __init__(self):
+        pass
+
+class Ec2Dummy:
+    class Instance:
+        def __init__(self,instance_id):
+            self.instance_id = instance_id
+            self.placement = {}
+            self.placement['AvailabilityZone'] = 'abc'
+
+        def load(self):
+            pass
+
+class EC2ClientDummy:
+    def __init__(self):
+        pass
+
+class S3Dummy:
+    class Bucket:
+        def __init__(self, name):
+            if name ==valid_container:
+                self.name = name
+                return
+            else:
+                raise Exception('Container not found')
```

Comment:
Again, it would be good to check the actual response from S3 and respond with similar object.